### ISSUE TITLE
React is not firing `onMouseLeave` events when using `onMouseLeave` inside `PopoverTrigger`

### ISSUE THREAD

**Issue by cheton** [cheton/styled-ui#588]
This issue was reported by Kira Wang (RD-CN)

---
在我使用Popover和Icon组件联动的时候，我的Icon组件中icon属性是变化的，这会导致某些情况下会出现PopoverContent内容弹出之后无法关闭的情况，所以想请教一下这是什么原因呢，下面是示例代码

```jsx
function Example() {
 const [iconHover, setIconHover] = React.useState(false);
  return (
    <>
      <Popover trigger="hover">
        <PopoverTrigger>
          <Icon
            onMouseEnter={() => setIconHover(true)}
            onMouseLeave={() => setIconHover(false)}
            icon={iconHover ? "info" : "info-o"}
           />
        </PopoverTrigger>
        <PopoverContent>
          Popover
        </PopoverContent>
      </Popover>
    </>
  );
}
```

当我鼠标快速滑动离开PopoverTrigger的情况下就会出现这种问题，如果鼠标慢速滑动离开PopoverTrigger就会正常。或者如果我的Icon组件的icon属性不变化的话，它也是正常的。
